### ISSUE TITLE
Segmentation fault: 11

### ISSUE THREAD

**Issue by nuttyartist** [gaogaotiantian/viztracer#307]
I get this error while running viztracer using this command `viztracer script.py`:

`Segmentation fault: 11`

Running on macOS Monetary 12.3 installed using pip.

**Comment by gaogaotiantian** [comment]
Could you share the script you were running? Also which version of `viztracer` and `python` you were using?

**Comment by nuttyartist** [comment]
I think it was an issue with Python 3.11. With Python 3.9 it now works.

**Comment by gaogaotiantian** [comment]
It your code segfaults without `viztracer` on Python 3.11 then yes it's a Python issue. However, if it only segfauls with `viztracer` on Python3.11, then it's still a `viztracer` issue as `viztracer` supports 3.11. In that case, I'd still check it out if you want to share some information.

**Comment by nuttyartist** [comment]
Oh ok. It only segfaults on Python 3.11. What can I provide you with? The script is part of a currently proprietary app, so it's not easy to share the full script.

**Comment by gaogaotiantian** [comment]
Ah, that's a bit tricky. It would be nice to have a minimal demo, like a small piece of shareable code to reproduce this issue, but I know that's extra workload on you and I'm not in the position to ask for it. If it's easy to have a small demo to reproduce it, I'll definitely take a look at it and try to figure out what's wrong. Otherwise, I'm not sure what kind of information would be helpful because asking you to figure out what segfaults is even more work.

So, if everything works for you know on 3.9 and you do not want to spend more time on `viztracer`, let's just leave this issue open and maybe in the future someone has a similar issue so they can refer to this.

**Comment by nuttyartist** [comment]
Well, I ran some tests and I'm sure what's the culprit now.

It's `import spacy`. This code reproduces the error. I'm using the latest version of spacy.

```python
import spacy

def main():
    print("hey")
    
if __name__ == "__main__":
    main()
```

**Comment by gaogaotiantian** [comment]
Thanks! This is super helpful. I can reproduce it locally now.

**Comment by gaogaotiantian** [comment]
Okay I took a deeper look into this. This is not a `viztracer` bug because I can reproduce it with the official `setprofile` function.

```python
import sys
def func(*args):
    frame = args[0]
    while frame:
        frame = frame.f_back

sys.setprofile(func)
import spacy
```

I think it's caused by the incompatibility between the new frame layout of 3.11 and `cython` or `thinc`. I have reported this in `spacy` repo https://github.com/explosion/spaCy/issues/12105.

As it's definitely not a `viztracer` specific bug, I'll close this issue.
